drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,412 @@
+	lis2dux12_md_t val = { .odr = cfg->odr, .fs = range };
+
+	err = lis2dux12_mode_set(ctx, &val);
+
+	if (!err) {
+		switch (range) {
+		default:
+		case LIS2DUX12_DT_FS_2G:
+			data->gain = lis2dux12_from_fs2g_to_mg(1);
+			break;
+		case LIS2DUX12_DT_FS_4G:
+			data->gain = lis2dux12_from_fs4g_to_mg(1);
+			break;
+		case LIS2DUX12_DT_FS_8G:
+			data->gain = lis2dux12_from_fs8g_to_mg(1);
+			break;
+		case LIS2DUX12_DT_FS_16G:
+			data->gain = lis2dux12_from_fs16g_to_mg(1);
+			break;
+		}
+	}
+
+	return err;
+}
+
+#define FOREACH_ODR_ENUM(ODR_VAL)                                                                  \
+	ODR_VAL(LIS2DUX12_DT_ODR_OFF, 0.0f)                                                        \
+	ODR_VAL(LIS2DUX12_DT_ODR_1Hz_ULP, 1.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_3Hz_ULP, 3.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_25Hz_ULP, 25.0f)                                                  \
+	ODR_VAL(LIS2DUX12_DT_ODR_6Hz, 6.0f)                                                        \
+	ODR_VAL(LIS2DUX12_DT_ODR_12Hz5, 12.50f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_25Hz, 25.0f)                                                      \
+	ODR_VAL(LIS2DUX12_DT_ODR_50Hz, 50.0f)                                                      \
+	ODR_VAL(LIS2DUX12_DT_ODR_100Hz, 100.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_200Hz, 200.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_400Hz, 400.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_800Hz, 800.0f)
+
+#define GENERATE_VAL(ENUM, VAL) VAL,
+
+static const float lis2dux12_odr_map[LIS2DUX12_DT_ODR_END] = {FOREACH_ODR_ENUM(GENERATE_VAL)};
+
+static int lis2dux12_freq_to_odr_val(const struct device *dev, uint16_t freq)
+{
+	const struct lis2dux12_config *cfg = dev->config;
+
+	/* constrain loop to prevent erroneous power mode/odr combinations */
+	size_t i = (cfg->pm != LIS2DUX12_OPER_MODE_LOW_POWER) ? LIS2DUX12_DT_ODR_6Hz
+							      : LIS2DUX12_DT_ODR_1Hz_ULP;
+	size_t len = (cfg->pm != LIS2DUX12_OPER_MODE_LOW_POWER) ? LIS2DUX12_DT_ODR_END
+								: LIS2DUX12_DT_ODR_6Hz;
+
+	while (i < len) {
+		if (freq <= lis2dux12_odr_map[i]) {
+			return i;
+		}
+		++i;
+	}
+
+	return -EINVAL;
+}
+
+static int lis2dux12_accel_config(const struct device *dev, enum sensor_channel chan,
+				  enum sensor_attribute attr, const struct sensor_value *val)
+{
+	switch (attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		return lis2dux12_set_range(dev, sensor_ms2_to_g(val));
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		LOG_DBG("%s: set odr to %d Hz", dev->name, val->val1);
+		int odr_val = lis2dux12_freq_to_odr_val(dev, val->val1);
+
+		if (odr_val < 0) {
+			LOG_ERR("frequency not unsupported or wrong operating mode.");
+			return odr_val;
+		}
+		return lis2dux12_set_odr(dev, odr_val);
```

Comment:
usually good practice to keep assignment next to validation, I know compliance is complaining but how about

```suggestion
		int odr_val;

		int odr_val = lis2dux12_freq_to_odr_val(dev, val->val1);
		if (odr_val < 0) {
			LOG_ERR("frequency not unsupported or wrong operating mode.");
			return odr_val;
		}
		
		LOG_DBG("%s: set odr to %d Hz", dev->name, val->val1);

				return lis2dux12_set_odr(dev, odr_val);
```

also, the LOG_ERR message should probably indicate `val->val1`

Reply:
Ok. I refactored it to be as shown in your suggestion.